### PR TITLE
stm32 button: use more idiomatic GPIO read

### PR DESCRIPTION
**Pull request by rohieb**
Using GPIOA_IDR directly looks a bit arcane, and the example can benefit from introducing gpio_get() and how to use it.

### CODE REVIEW

**Comment by rohieb** [comment]
Forgot to clean up the unused variable, amended and updated the patch.

File: examples/stm32/f1/stm32-h103/button/button.c [hunk 1]
```diff
@@ -62,8 +60,7 @@ int main(void)
 	while (1) {
 		gpio_toggle(GPIOC, GPIO12);
 
-		exti_line_state = GPIOA_IDR;
-		if ((exti_line_state & (1 << 0)) != 0) {
+		if ((gpio_get(GPIOA, GPIO0) & GPIO0) == 1) {
```
Comment (karlp) [inline examples/stm32/f1/stm32-h103/button/button.c]: This is redundant.  gpio_get is implemented as gpio_port_read(port) & pin.  Simply
```C
if (gpio_get(GPIOA, GPIO0)) {
```
is sufficient.  The == 1 is misleading as well, it only works for gpio0.

File: examples/stm32/f0/stm32f0-discovery/button/button.c [hunk 2]
```diff
@@ -62,7 +62,7 @@ int main(void)
 		gpio_toggle(GPIOC, GPIO8);
 
 		/* Upon button press, blink more slowly. */
-		if (gpio_get(GPIOA, GPIO0)) {
+		if ((gpio_get(GPIOA, GPIO0) & GPIO0) == 1) {
```
Comment (karlp) [inline examples/stm32/f0/stm32f0-discovery/button/button.c]: No, this is worse, and likely to lead to errors if you try and change to say, GPIO5

**Comment by karlp** [comment]
Thanks for taking the time to properly clean it up across all examples, but please don't use the style you've replaced it with.

**Comment by rohieb** [comment]
Oh right, it seems I was misreading the documentation for `gpio_get()`. After re-reading it, the STM32F0 example looks good to me. I've updated the patch so the other examples follow its style.

**Comment by karlp** [comment]
Thanks for this!

**Pull request merged**